src/main.rs

```diff
@@ -31,10 +33,15 @@ async fn rocket() -> _ {
 
     let config = Config::new();
     let database = Database::new(&config).await;
-    let services = Services::new(&config, database);
+    let database = Arc::new(database);
+    let services = Services::new(&config, Arc::clone(&database));
     let services = Arc::new(services);
     let graphql_schema = Schema::build(Query::default(), Mutation::default(), EmptySubscription)
         .data(Arc::clone(&services))
+        .data(DataLoader::new(
+            UserLoader::new(Arc::clone(&database)),
+            rocket::tokio::spawn,
+        ))
```

Comment:
Ever time we want to instantiate a loader a Tokio task is spawned. We must provide the Spawner to `DataLoader` so instances are spawned during initialization.